kadmin/check.c

```diff
@@ -90,7 +90,7 @@ do_check_entry(krb5_principal principal, void *data)
 	    krb5_warnx(context,
 		       "Principal %s enctype %d, wrong length: %lu\n",
 		       name, princ.key_data[i].key_data_type[0],
-		       (unsigned long)princ.key_data[i].key_data_length);
+		       (unsigned long)princ.key_data[i].key_data_length[0]);
```

Comment:
could you remove the cast to `unsigned long` and change the format string to use `%hu` instead of `%lu` as the type is `int16_t`?   The cast was masking the prior use of the pointer value as an integer.

Reply:
I'll change the `%lu` to `%d`, since `princ.key_data[i].key_data_length[0]` is an `int16_t` that will get promoted to `int` anyways.